### PR TITLE
Add Counter to Helpers

### LINKED ISSUE #4998: Add counter to the helper UI
With https://github.com/home-assistant/home-assistant-polymer/pull/4940 we have a UI to add and manage the `input_*` domains.

The same should be done for the `counter` domain. There is no support on the backend yet, issue to track that: https://github.com/home-assistant/home-assistant/issues/32226

### PR DESCRIPTION
**Pull request by danimart1991**
## Proposed change

Add Counter to Helpers UI.

image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
counter:
```

## Additional information

- This PR fixes or closes issue: fixes #4998 
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by danimart1991** [comment]
I'm currently working on this. A few questions:

- I have problems saving the value of the `restore bool` value while using a Switch. Maybe I forgot something or I'm using the wrong component?
- Actually `default_config` component doesn't include `Counter` so I think is necessary an extra step to include it.

**Comment by bramkragten** [comment]
Yes, `counter` should be added to `default_config` in the core repo.

File: src/panels/config/helpers/forms/ha-counter-form.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+  @internalProperty() private _name!: string;
+
+  @internalProperty() private _icon!: string;
+
+  @internalProperty() private _maximum?: number;
+
+  @internalProperty() private _minimum?: number;
+
+  @internalProperty() private _restore?: bool;
```
Comment (bramkragten) [inline src/panels/config/helpers/forms/ha-counter-form.ts:36]:
```suggestion
  @internalProperty() private _restore?: boolean;
```

File: src/panels/config/helpers/forms/ha-counter-form.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        }
+        .row {
+          margin-top: 8px;
+          color: var(--primary-text-color);
+          display: flex;
+          align-items: center;
+        }
+        ha-paper-dropdown-menu {
+          display: block;
+        }
```
Comment (bramkragten) [inline src/panels/config/helpers/forms/ha-counter-form.ts:199]: Not used?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**